Added testiIfBookmarkedMoviesIsEmpty, testAddNewItemToBookmarkedMovie…

softvision-valentinboca opened the pull request:
…s, testShareItem tests

softvision-valentinboca committed:
Added testiIfBookmarkedMoviesIsEmpty, testAddNewItemToBookmarkedMovies, testShareItem tests

turcumadalina commented on app/src/androidTest/java/com/popularMovies/tests/BookmarkTests.java:
```diff
@@ -0,0 +1,36 @@
+    @Test
+    public void testiIfBookmarkedMoviesIsEmpty() throws Exception {
+        NavDrawer.bookmarkedMovies();
+        Helpers.isItemDisplayed("NO BOOKMARKS ADDED");
```
Textele le adaugam in clasa Strings :)

turcumadalina commented:
Arata super bine, se vede progresul, dar observ ca nu ai lucrat cu screen-uri. :)

softvision-valentinboca committed:
used Strings, Screens. Created booleans for assertions.
Created testPlayItem (with 3rd pary app) test

turcumadalina commented on app/src/androidTest/java/com/popularMovies/screens/BookmarkMovie.java:
```diff
@@ -0,0 +1,26 @@
+public class BookmarkMovie {
+    public static boolean isNoBookmarksAdded() throws Exception{
+        try{
+            onView( allOf( withText(R.string.bookmarks_empty), isCompletelyDisplayed())).check(matches(isDisplayed()));
```
Foloseste "checkIfUIObjectIsVisible"

turcumadalina commented on app/src/androidTest/java/com/popularMovies/screens/NowPlayingMovies.java:
```diff
@@ -16,4 +23,13 @@ public static void clickAMovieNowPlayingMovies(String text) throws Exception {
+
+    public static boolean isNowPlayingMovies() throws Exception{
+        try{
+            onView( allOf( withText(Strings.NOW_PLAYING_MOVIES), isCompletelyDisplayed())).check(matches(isDisplayed()));
```
Foloseste "checkIfUIObjectIsVisible"

turcumadalina commented on app/src/androidTest/java/com/popularMovies/tests/BookmarkTests.java:
```diff
@@ -14,11 +17,13 @@
     public void testiIfBookmarkedMoviesIsEmpty() throws Exception {
         NavDrawer.bookmarkedMovies();
-        Helpers.isItemDisplayed("NO BOOKMARKS ADDED");
+        Assert.assertTrue(Strings.NO_BOOKMARKS, BookmarkMovie.isNoBookmarksAdded());
```
Sigur trece testul? :-?

Ce reprezinta "Strings.NO_BOOKMARKS"? Acel parametru trebuie sa fie mesajul de eroare care o sa fie returnat in caz ca pica assert-ul. ex: "The title should be displayed."

softvision-valentinboca committed:
done

softvision-valentinboca committed:
Created testRedirectingToHomePage test. Created a method for scrolling, searching and clicking on an item from the list

turcumadalina commented on app/src/androidTest/java/com/popularMovies/screens/NowPlayingMovies.java:
```diff
@@ -0,0 +1,25 @@
+public class NowPlayingMovies {
+    public static void clickAMovieNowPlayingMovies(String text) throws Exception {
+        onView(withText(text)).perform(click());
+        Helpers.isItemDisplayed(text);
```
lista asta nu verifica nimic in contextul asta

turcumadalina commented on app/src/androidTest/java/com/popularMovies/tests/BookmarkTests.java:
```diff
@@ -0,0 +1,59 @@
+        Movie.bookmarkMovie();
+        pressBack();
+        NavDrawer.bookmarkedMovies();
+        Helpers.isTextDisplayed(  "6.0/10", R.id.vote_text);
```
Nici asta

softvision-valentinboca committed:
made the right assertions (hope it's better :-s)

softvision-valentinboca committed:
added the testIfAnActorIsDisplayed test.
created a new method for clicking a view inside a recyclerview

softvision-valentinboca committed:
created testAddNewTvShowToBookmarkedTvShows
tried to get rid of most of the hardcoded elements from the tests

softvision-valentinboca committed:
created testIfSimiliarMoviesAreDisplayed; testAddNewTvShowToBookmarkedTvShows tests
created a method for scrolling to a certain element from a listview/recyclerview

softvision-valentinboca committed:
added a new method for Pull to Refresh function

turcumadalina commented:
Wow  👍  Cate chestii faine ai facut! Gooood job!

softvision-valentinboca committed:
created a method for scrolling horizontally through a RecyclerView

softvision-valentinboca committed:
added a method for ToastMatcher (method used from Ioana's PR, just for practice)
